src/JLD.jl

```diff
@@ -1209,7 +1209,7 @@ FileIO.save(f::File{format"JLD"}, value...; kwargs...) = error("Must supply a na
 # load with just a filename returns a dictionary containing all the variables
 function FileIO.load(f::File{format"JLD"})
     jldopen(FileIO.filename(f), "r") do file
-        (String => Any)[var => read(file, var) for var in names(file)]
+        Dict{String,Any}(var => read(file, var) for var in names(file))
```

Comment:
So after seeing the details, this seems to be the problem. This was done on mac and julia 0.5(Commit d98f2c0). This seems fine on 0.5, but errors on 0.4 inspite of using `@compat`. I made this change based on the deprecation warning, 
```
WARNING: deprecated syntax "[a=>b for (a,b) in c]".
Use "Dict(a=>b for (a,b) in c)" instead.
```